### ISSUE TITLE
ConnectionLocator cannot be used with PDO Persistent Connections

### ISSUE THREAD

**Issue by dbraff** [atlasphp/Atlas.Pdo#11]
Hi, I was looking into integrating this library with an existing application and while running a small test I found an a compatibility issue between the `ConnectionLocator` class and use of `PDO::ATTR_PERSISTENT`. It seems that the `ConnectionLocator->newConnection()` method always calls the `Connection->logQueries()` method, which in turn sets the attribute `PDO::ATTR_STATEMENT_CLASS`. According to the PDO documentation (https://www.php.net/manual/en/pdo.setattribute.php), these two attribute settings cannot be used together:

> PDO::ATTR_STATEMENT_CLASS: Set user-supplied statement class derived from PDOStatement. Cannot be used with persistent PDO instances. Requires array(string classname, array(mixed constructor_args)).

This would also be an issue if you instantiated a `Connection` object and called it's `logQueries()` method, but in the case of the `ConnectionFactory` it happens automatically.

```
<?php

use Atlas\Pdo\ConnectionLocator;

require_once __DIR__ . '/vendor/autoload.php';

...

$connectionLocator = ConnectionLocator::new(
    $dsn,
    $username,
    $password,
    [
        PDO::ATTR_PERSISTENT => true,
    ]
);

$connection = $connectionLocator->getDefault();
```
Running the above sample code will produce the following exception:

> Fatal error: Uncaught PDOException: SQLSTATE[HY000]: General error: PDO::ATTR_STATEMENT_CLASS cannot be used with persistent PDO instances in ...

I think this is an excellent package and would love to be able to use it in my organization. I would be happy to submit a PR for this issue or alternatively to add to the documentation so it is known that persistent connections are not compatible with this package if that is the desired approach.

Thank you,
Daniel Braff

**Comment by pmjones** [comment]
Thanks for the very good problem report!

> I would be happy to submit a PR for this issue or alternatively to add to the documentation so it is known that persistent connections are not compatible with this package if that is the desired approach.

The former would be preferred, since it would actually solve your problem, but the latter would also be welcome.

**Comment by dbraff** [comment]
Thank you for the quick response. I'll work on a PR to fix the issue hopefully this evening.

**Comment by pmjones** [comment]
Fixed by #12
